src/Adapter/Form/ChoiceProvider/ShopTreeChoiceProvider.php

```diff
@@ -58,6 +58,10 @@ public function getChoices()
             foreach ($shops as $shopId) {
                 $shop = Shop::getShop($shopId['id_shop']);
 
+                if (!$shop['id_shop']) {
```

Comment:
why is it needed? does it have something to do with permissions?

Reply:
it's needed because the code in line 59 `$shop = Shop::getShop($shopId['id_shop']);` will not retrieve the shop data if the employee doesn't have access to it, in such case an empty checkbox without a label or value would be displayed in the shop tree: https://prnt.sc/mrecke

Reply:
it could also be useful to check for `false` i think, just in case.

```diff
- if (!$shop['id_shop'])
+ if (false === $shop || !$shop['id_shop'])
```

Reply:
actually I think one `false` check is enough - in legacy code it returns either `false` or the array with shop data. 
`if (false === $shop)`
I'll fix that.

Reply:
```suggestion
                if (empty($shop['id_shop'])) {
```
Otherwise